test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -583,8 +613,27 @@ private NodeAndClient buildNode(int nodeId, long seed, Settings settings, Versio
         return new NodeAndClient(name, node);
     }
 
-    private String buildNodeName(int id) {
-        return nodePrefix + id;
+    private String buildNodeName(int id, Settings settings) {
+        String prefix = nodePrefix;
+        prefix = prefix + getRoleSuffix(settings);
+        return prefix + id;
+    }
+
+    /** returns a suffix string based on the node role. If no explicit role is defined, the suffix will be empty */
+    private String getRoleSuffix(Settings settings) {
+        String suffix = "";
+        if (Node.NODE_MASTER_SETTING.exists(settings) && Node.NODE_MASTER_SETTING.get(settings)) {
+            suffix = suffix + "m";
```

Comment:
why do we need c for coord only again? The reason why its abbreviation is empty is that coordinating requests is the least a node can do, and every node will do that.